src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs

```diff
@@ -3072,6 +3072,10 @@
     ("SveConditionalSelect.template",     new Dictionary<string, string> { ["TestName"] = "Sve_ConditionalSelect_uint",                                                                            ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "ConditionalSelect",                                                    ["RetVectorType"] = "Vector",    ["RetBaseType"] = "UInt32",  ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "UInt32", ["Op2VectorType"] = "Vector",    ["Op2BaseType"] = "UInt32", ["Op3VectorType"] = "Vector",    ["Op3BaseType"] = "UInt32", ["LargestVectorSize"] = "64",  ["NextValueOp1"] = "TestLibrary.Generator.GetUInt32()",       ["NextValueOp2"] = "TestLibrary.Generator.GetUInt32()",        ["NextValueOp3"] = "TestLibrary.Generator.GetUInt32()",                         ["ValidateIterResult"] = "(firstOp[i] != 0 ? (result[i] != secondOp[i]) : (result[i] != thirdOp[i]))",}),
     ("SveConditionalSelect.template",     new Dictionary<string, string> { ["TestName"] = "Sve_ConditionalSelect_ulong",                                                                           ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "ConditionalSelect",                                                    ["RetVectorType"] = "Vector",    ["RetBaseType"] = "UInt64",  ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "UInt64", ["Op2VectorType"] = "Vector",    ["Op2BaseType"] = "UInt64", ["Op3VectorType"] = "Vector",    ["Op3BaseType"] = "UInt64", ["LargestVectorSize"] = "64",  ["NextValueOp1"] = "TestLibrary.Generator.GetUInt64()",       ["NextValueOp2"] = "TestLibrary.Generator.GetUInt64()",        ["NextValueOp3"] = "TestLibrary.Generator.GetUInt64()",                         ["ValidateIterResult"] = "(firstOp[i] != 0 ? (result[i] != secondOp[i]) : (result[i] != thirdOp[i]))",}),
 
+    ("SveMasklessSimpleVecOpTest.template",        new Dictionary<string, string> { ["TestName"] = "Sve_ConvertToInt32_int_float",                                                                          ["Isa"] = "Sve",           ["LoadIsa"] = "Sve",     ["Method"] = "ConvertToInt32",                                                       ["RetVectorType"] = "Vector",    ["RetBaseType"] = "Int32",   ["Op1VectorType"] = "Vector",    ["Op1BaseType"] = "Single",                                                                                                                           ["LargestVectorSize"] = "64", ["NextValueOp1"] = "TestLibrary.Generator.GetSingle()",                                                                                                                                                      ["ValidateIterResult"] = "Helpers.ConvertToInt32(firstOp[i]) != result[i]", ["GetIterResult"] = "Helpers.ConvertToInt32(leftOp[i])"}),
```

Comment:
we should use `SveSimpleVecOpTest.template` here.

Reply:
We ended up needing a new test template to handle the fact that these APIs' return types are different from their input types, so the `ConditionalSelect` scenarios' first and third operands need to match the API's return type. The new template is otherwise functionally equivalent to `SveSimpleVecOpTest.template`, though we removed the `ConditionalSelect_Op1` scenario since `ConditionalSelectScenario(_mask, _fld1, _fld1);` doesn't satisfy the new method signature.

Reply:
Removing `ConditionalSelect_Op1 ()` makes sense, but apart from that, do you think we could have fixed `SveSimpleVecOpTest.template` to have `RetType` wherever necessary and then just update the value of it to be same as `Op1BaseType` for other tests and having a different one for `Convert` API?

Reply:
That makes sense to me, though would you be ok with us getting rid of `ConditionalSelect_Op1` for all existing tests using `SveSimpleVecOpTest.template`? If we aren't losing anything valuable from that, then I can merge the two templates in a follow-up PR after this one is merged.

Reply:
We definitely need `ConditionalSelect_Op1`, but was wondering if other than it can be fixed or not. We can probably merge under the condition of `if (typeof({Op1BaseType}) == typeof({RetBaseType})` or something.

Reply:
I see, I can add that check

Reply:
`if (typeof({Op1BaseType}) == typeof({RetBaseType})` - I'm not sure if this would work at the compile of the .cs file. For the mismatching cases, the code inside the `if` block would have mismatching types, and probably get type mismatch errors. Would work if Rosyln just optimises it all away. Alternatively, you can probably add a lot of casts.

Reducing the number of templates is good.

While your cleaning this up, the conditional tests it lots of places use `{Op1VectorType}<{RetBaseType>`. These should ideally be `{RetVectorType}<{RetBaseType>` or `{Op1VectorType}<{Op1BaseType>`. Doesn't matter which as they'll be the same.